main.py

```diff
@@ -60,4 +81,11 @@ def main(sel, fname):
 
 if __name__ == '__main__':
     args = parser.parse_args()
-    main(args.sel, args.fname)
+
+    num_gpus = len(args.devices)
+    processes = []
+    for rank in range(num_gpus):
```

Comment:
Why the for loop? I think you only need to call `mp.spawn` once, right? With `nprocs=num_gpus` it will spawn `nprocs` processes already, one per gpu

Reply:
and yes, no need for the loop here as well